### PR TITLE
Update MSI workflow

### PR DESCRIPTION
**Pull request by YingChen1996**
This pull request includes changes to improve the build process and package installation for the project. The most important changes include modifying the build_msi_installer.yml file to specify the correct requirements file for installing Python packages and adding a new input parameter to specify a requirements file during the setup step.

Build process improvements:

* <a href="diffhunk://#diff-0d867135d0d5e125130c57972db2f426e7a20a27fe334ec7185497eb54570d3aR51-R60">`.github/workflows/build_msi_installer.yml`</a>: Modified the build_msi_installer.yml file to specify the correct requirements file for installing Python packages, remove unnecessary environment variables, and add steps to download a JSON file from Azure Blob Storage and compare the version with the input. <a href="diffhunk://#diff-0d867135d0d5e125130c57972db2f426e7a20a27fe334ec7185497eb54570d3aR51-R60">[1]</a> <a href="diffhunk://#diff-0d867135d0d5e125130c57972db2f426e7a20a27fe334ec7185497eb54570d3aL21-L22">[2]</a> <a href="diffhunk://#diff-0d867135d0d5e125130c57972db2f426e7a20a27fe334ec7185497eb54570d3aR76-R79">[3]</a> <a href="diffhunk://#diff-0d867135d0d5e125130c57972db2f426e7a20a27fe334ec7185497eb54570d3aR134-R154">[4]</a>

Package installation improvements:

* <a href="diffhunk://#diff-40a9881b0d7d3e1185fa0078f841940c7766980271b9ef9abe103205f34ee669R10-R13">`.github/actions/step_sdk_setup/action.yml`</a>: Added a new input parameter to specify a requirements file for installing Python packages during the setup step and modified the run command to use the specified requirements file. Also added an echo command to print the list of installed packages before building the wheel. <a href="diffhunk://#diff-40a9881b0d7d3e1185fa0078f841940c7766980271b9ef9abe103205f34ee669R10-R13">[1]</a> <a href="diffhunk://#diff-40a9881b0d7d3e1185fa0078f841940c7766980271b9ef9abe103205f34ee669L38-R42">[2]</a> <a href="diffhunk://#diff-40a9881b0d7d3e1185fa0078f841940c7766980271b9ef9abe103205f34ee669L24-R28">[3]</a>

Package requirements:

* <a href="diffhunk://#diff-24103fa518e86ae294d853e2d1948ea60f010b2c17606d6e74943a4c8e74a2c8R1-R2">`src/promptflow/msi_requirements.txt`</a>: Added `setuptools` and `wheel` as required Python packages in `src/promptflow/msi_requirements.txt` for building and distributing Python packages.# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/build_msi_installer.yml [hunk 1]
```diff
@@ -126,18 +128,27 @@ jobs:
...
+          az storage blob download --account-name promptflowartifact --container-name msi-installer --name latest_version.json --file downloaded_version.json
+          $downloaded_version=$(python -c "import json; new_version = json.load(open('downloaded_version.json'))['promptflow']")
+          echo "::set-output name=downloaded_version::downloaded_version"
+
       - name: Check if version input is valid and upload JSON file
         run: |
           $version = "${{ steps.get-version.outputs.version }}"
-          if ($version -like '1.*') {
+          $downloaded_version = "${{ steps.download-json.outputs.downloaded_version }}"
+          if ($version -like '1.*' -and $version -gt $downloaded_version){
```
Comment (wangchao1230) [inline .github/workflows/build_msi_installer.yml:142]: -gt cannot detect case like 
"1.10.1" -gt "1.9.0"

Comment (YingChen1996) [reply .github/workflows/build_msi_installer.yml:142]: fixed

Comment (wangchao1230) [reply .github/workflows/build_msi_installer.yml:142]: what's the fix here?

Comment (YingChen1996) [reply .github/workflows/build_msi_installer.yml]: image

**Review by wangchao1230 — DISMISSED** [review]

**Review by elliotzh — DISMISSED** [review]

**Review by elliotzh — APPROVED** [review]

**Pull request merged**